modules/meshcloud-replicator-service-principal/module.tf

```
@@ -187,7 +191,7 @@ resource "azuread_app_role_assignment" "meshcloud_replicator-user" {
 // Assign it to the specified scope
 //---------------------------------------------------------------------------
 resource "azurerm_policy_definition" "privilege_escalation_prevention" {
-  name                = "meshcloud-privilege-escalation-prevention"
+  name                = "meshcloud-privilege-escalation-prevention-${local.spp_hash}"
```

Can we use the scope in the name? Makes it easier to read.
Like this:
```suggestion
  name                = "meshcloud-privilege-escalation-prevention-${var.custom_role_scope}"
```

Same question for assignment

---

Unforunately this will make it break if another service principal was created in a tenant. It has to be unique per SP, and the only idea was to hash the SP name. Using the SP name itself also doesn't work for the assignment since it has a max length of 24